### ISSUE TITLE
glance_image not idempotent

### ISSUE THREAD

**Issue by peterjenkins1** [ansible/ansible-modules-core#820]
glance_image not idempotent which results in one duplicate image for each playbook run.

This looks like a problem with https://github.com/ansible/ansible-modules-core/blob/3799ca48ea34bb3e5bc4c200e9d3aa9d357c016e/cloud/openstack/glance_image.py#L211

Specifically 'client.images.list()' in v1 of the glance API seems to only return a list of images from the current tenant, even for admins. Changing the module to use the v2 api for this call would fix this issue, but using v2 for the whole module will break other things.

**Comment by robynbergeron** [comment]
@emonty @shrews @juliakreger @j2sol @rcarrillocruz - FYI (sorry for the sudden flood, i'm doing some labelling / research and uncovering all sorts of untouched historical things)

**Comment by emonty** [comment]
glance_image is old, broken and deprecated. I highly recommend moving to os_image.

**Comment by ansibot** [comment]
@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by Shrews** [comment]
Please use os_image as this module is deprecated.

wontfix
